README.md

```diff
@@ -12,39 +12,134 @@ Structure
 ---------
 
   * `data`
-    * `tests.json`—tests, initially converted from Rust file
-      `map_model/src/make/initial/lane_specs.rs` of A/B Street project.  Code is
-      under Apache-2.0 License.
+    * `tests.json`—tests, initially converted from Rust file.
+      `spec-lanes.json`—json specification.
   * `kotlin`—[Kotlin implementation](#kotlin).
   * `python`—[Python 3.9 implementation](#python).
   * `rust`—[Rust implementation](#rust).
 
-Example
--------
+Lane Definition
+---------------
 
-Input JSON file with road OpenStreetMap tags:
+It is important to determine a strict definition for a lane.
+An OpenStreetMap way, usually a `highway=*`, can in some cases be divided into lanes,
+where each lane extends from the start to the end of the way.
+A lane's direction of travel is not necessarily the same as the direction of the way.
+A lane represents a continuous area with no formal separation of vehicles.
+Lanes are often separated by something visible like painted lines or raised curbs,
+but these are not necessary for lanes to exist.
+
+
+### Example 1 ###
+
+A road with 5 marked lanes.
+
+![Example 1](https://upload.wikimedia.org/wikipedia/commons/0/0e/Gr%C3%BCnbergstra%C3%9Fe_2.JPG)
 
 ```json
-{
-    "lanes": "2",
-    "oneway": "yes",
-    "sidewalk": "both",
-    "cycleway:left": "lane"
-}
+[
+    {"type": "travel_lane", "direction": "backward"},
+    {"type": "separator", "markings": [{"style": "dashed_line"}]},
+    {"type": "travel_lane", "direction": "backward"},
+    {"type": "separator", "markings": [{"style": "solid_line"}, {"style": "solid_line"}]},
+    {"type": "travel_lane", "direction": "forward"},
+    {"type": "separator", "markings": [{"style": "dashed_line"}]},
+    {"type": "travel_lane", "direction": "forward"},
+    {"type": "separator", "markings": [{"style": "broken_line"}]},
+    {"type": "travel_lane", "direction": "forward"},
+]
 ```
 
-Output lane specifications from left to write:
+Note:
+
+- there is no sidewalk nor verge for pedestrians to walk here.
+- lane separators are listed.
+- we omit widths for simplicity. (TODO: if a guess is needed, e.g. on an average lane width, where is this guess made?)
```

Comment:
That's one of the next things I'm interested in adding. The Berlin group behind [this map](https://supaplexosm.github.io/strassenraumkarte-neukoelln/posts/2021-12-31-micromap-update) has been adding cycleway and separator widths in OSM.

I think this guess could live in `osm2lanes`. Something like
```
width: 2.0,    // always meters, per the spec
width_source: "osm | assumption"
```

Defaults probably vary by `highway` type and region. We can think about some "profiles" in `Config`